### ISSUE TITLE
Runtime crash if using in a project that already uses ASIO

### ISSUE THREAD

**Issue by sasmaster** [rpclib/rpclib#179]
The system is WINDOWS10  ,64bit Project, VS2015.

Here is what I have. I have an executable project.Some of the classes in it use standalone ASIO API.
I linked the rpc.lib to that project,which I built with CMAKE. And when I launch server example code I am getting crash on this method:

  ```
      void lock()
       {
          ::EnterCriticalSection(&crit_section_);
       }
```

Inside  asio/detail/ win_mutex,hpp.

The whole object is a mess.Almost nothing looks initialized in debugger. I thought maybe I have wrong configs on the project level. What I did was - created clean new exe project.Linked to it the rpc.lib. And it worked ok. then I moved all the properties from my initial project into the new one,including all the preprocessor defs,includes,linker settings etc. And the new one still worked. And only when I imported the sources from the original project into the new one, then I received the same crash. Now,the only familiar situation I encountered was when there were issues related to different compilers,different versions of compiled sources ,or some other differences in project setup which impacted the binary level of the code. Here I can't understand why I am getting it. My sources use Websocketpp lib and standalone ASIO libs. Websocketpp uses my version of standalone ASIO lib.Now when I built the rpc.lib I linked it with the same ASIO lib I use in my project. Also,in the tests I removed all the inits to make sure no instance of some class can screw the whole thing with some memory error. So the only thing I got now inside main.cpp is this:

     ```
    #include <rpc/server.h>

int main()
{
	rpc::server srv(2354);

	srv.bind("echo", [](std::string const& s) {
		return s;
	});

	srv.run();
    return 0;
}
```
Now,of course the other classes get compiled into the final exe because their sources are still included.And as I said,if I don't include those source then rpc lib code works.

**Comment by sasmaster** [comment]
Yeah, if I remove classes where there is 'raw' ASIO API, rpc lib works. How can I solve this?

**Comment by sztomi** [comment]
No easy solution yet, unfortunately, but I definitely want to make the bundled dependencies optional - it was a mistake to so tightly integrate with them. The following issue tracks this: #155 

The *hard* solution is to get rid of the bundled asio headers and compile with what you are using. You can change the RPCLIB_ASIO macro to `asio` or `boost::asio`.

**Comment by sasmaster** [comment]
Yeah,that's what I am probably going to do for now. Will be waiting for your update. Thanks!

**Comment by sasmaster** [comment]
Hi again. After spending some time I figured out the following problem. (replacing your asio macro didn't help,btw)

This MACRO screwes everything :  

_WIN32_WINNT=0x0501


I am not sure why you need it. But if I add it to my project then the above mentione crash disappears .
